blocks/table/src/app.tsx

```diff
@@ -0,0 +1,305 @@
+
+                setSelection(emptySelection);
+              }}
+              className={styles.danger}
+            >
+              Delete
+            </div>
+          </>
+        </div>
+      )}
+      <Grid
+        rowMarkerWidth={32}
+        rows={rows.length}
+        columns={columns}
+        rightElement={
+          readonly || hideHeaderRow ? null : (
+            <div className={styles.addColumnButton} onClick={addNewColumn}>
+              Add a Column +
+            </div>
+          )
+        }
+        getRowThemeOverride={isStriped ? getRowThemeOverride : undefined}
+        onPaste
+        onHeaderMenuClick={handleHeaderMenuClick}
+        onCellsEdited={handleCellsEdited}
+        rightElementProps={{ fill: true }}
+        trailingRowOptions={{
+          hint: "New row...",
+          sticky: true,
+          tint: true,
+        }}
+        onRowAppended={
+          readonly
+            ? undefined
+            : () => {
+                /**
+                 * @todo this should be async, but making it async makes grid place the overlay with a weird offset
+                 * needs debugging
+                 */
+                addNewRow();
+              }
+        }
+        headerHeight={hideHeaderRow ? 0 : ROW_HEIGHT}
+        rowMarkers={hideRowNumbers ? "none" : readonly ? "number" : "both"}
+        rowSelectionMode="multi"
+        getCellContent={([colIndex, rowIndex]) => {
+          const key = columns[colIndex]?.id;
+
+          if (!key) {
+            return {
+              kind: GridCellKind.Text,
+              displayData: "",
+              data: "",
+              allowOverlay: false,
+            };
+          }
+
+          const value = rows[rowIndex]?.[key] ?? "";
+
+          return {
+            kind: GridCellKind.Text,
+            displayData: String(value),
+            data: String(value),
+            allowOverlay: !readonly,
+          };
+        }}
+        gridSelection={selection}
+        onGridSelectionChange={(newSelection) => setSelection(newSelection)}
+      />
+
+      {!!headerMenu &&
+        renderLayer(
+          <HeaderMenu
+            key={headerMenu.col}
+            layerProps={layerProps}
+            title={columns[headerMenu.col]?.title ?? ""}
+            onDelete={() => {
+              updateEntity({
+                [localColumnsKey]: localColumns.filter(
+                  (_, index) => index !== headerMenu.col,
+                ),
+              });
+              // delete column here
+              setHeaderMenu(undefined);
+            }}
+            onClose={() => setHeaderMenu(undefined)}
+            updateTitle={(title) => {
+              updateEntity({
+                [localColumnsKey]: localColumns.map((col, index) =>
+                  index === headerMenu.col
+                    ? { ...col, [columnTitleKey]: title }
+                    : col,
+                ),
+              });
+              // update column title here
```

Comment:
what's this for? is it a `@todo`?
```suggestion
              // update column title here
```